plonky2x/core/src/frontend/hash/poseidon/poseidon256.rs

```diff
@@ -115,7 +115,11 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
 
                 let hash_byte_vec = bit_list
                     .chunks(8)
-                    .map(|chunk| ByteVariable(array![i => BoolVariable::from(chunk[i].target); 8]))
+                    .map(|chunk| {
+                        ByteVariable::from_variables_unsafe(
```

Comment:
Should we also make `ByteVariable` safer by adding the `_private()` field as well? I think a separate PR might be best for that as well

Reply:
Actually, `from_variables_unsafe` is not needed here.  It's underlying elements are guaranteed to be booleans, so there is it's safe.  Will modify that.

Also, since ByteVariable's underlying variables are `BoolVariables`, it should be "safe" for someone to instantiate it directly.  The general principle that I did was to implement the `_private()` pattern for variables that could be be overflowed, and allow direct creation for all other ones.

This is actually somewhat similar to the principle used within plonky2.